File: yoroflow-ui/src/app/designer-module/task-flow/model/task-node.ts

import { NodePosition } from '@swimlane/ngx-graph';
import { TaskProperty } from '../../task-property/task-property-vo';

export interface TaskNode {
  key: string;
  taskType: string;
  label: string;
  taskProperty?: TaskProperty;
  position?: NodePosition;
}

File: yoroflow-service/src/main/java/com/yorosis/yoroflow/models/TaskNode.java
package com.yorosis.yoroflow.models;

import com.yorosis.yoroflow.models.decision.Condition;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TaskNode {
	private String key;
	private String taskType;
	private String label;
	private ProcessDefTaskPropertyVO taskProperty;
	private NodePosition position;
	private Condition condition;
}
